static/app/views/dashboardsV2/widgetBuilder/widgetBuilder.tsx

```diff
@@ -173,7 +173,12 @@ function WidgetBuilder({
     location.query.defaultWidgetQuery
   );
 
-  const isEditing = defined(widgetIndex);
+  const isEditing = defined(widgetIndex) && widgetIndex !== NEW_ID;
```

Comment:
I think `widgetIndex` is breaking for the Add to Dashboard from Discover flow right now. The `/dashboards/new/widget/new/` route is hardcoded in routes which means the `new` values for `widgetIndex` doesn't actually get picked up in params.

Reply:
yes, Ed is right... new will never be available in the `widgetIndex` as it is part of the router, but I have tested it locally and it doesn't seem to break for discover, since its widgets coming from discover are not being edited but added. In any case, should we update this logic?

Reply:
Good point, I realized eventually that they were separate routes and then could be undefined. Touched base with Ed and confirmed that he had a version from before I put `defined(widgetIndex)` back. I removed the assertion where it checks `widgetIndex !== NEW_ID`